infra/gravity/cluster_failover.go

```diff
@@ -0,0 +1,67 @@
+package gravity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+)
+
+// Failover isolates the current leader node and elects a new leader node.
+func (c *TestContext) Failover(nodes []Gravity) error {
+	// TODO: Configure timeouts
+	ctx, cancel := context.WithTimeout(c.ctx, c.timeouts.Status)
+	defer cancel()
+
+	oldLeader, err := getLeaderNode(ctx, nodes)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	c.Logger().Info(fmt.Sprintf("oldLeader=%v", oldLeader))
+
+	if err := oldLeader.Disconnect(ctx); err != nil {
+		return trace.Wrap(err, "failed to disconnect from cluster")
+	}
+
+	newLeader, err := getLeaderNode(ctx, nodes)
```

Comment:
This seems like it operates too quickly after Disconnect is called. Because the network is getting blocked, it's going to take the rest of the cluster some amount of time to realize the old leader has expired, and elect a new leader. So you'll need to wait/watch for the leadership change, and if it doesn't change after some timeout, then fail the test.